### PR TITLE
#1 Add label_list options to preprocess.py

### LINKED ISSUE #1: Add ```label_list``` options to ```preprocess.py```
Currently, we generate and filter the label_list in ```preprocess.py```. We should generate the list and parse it externally, then pass the label_list to ```preprocess.py``` for actual data filtering.

### PR DESCRIPTION
**Pull request by yyitingli**
#1 Add label_list options to preprocess.py

### CODE REVIEW

**Review by kamilkrukowski — CHANGES_REQUESTED** [review]
Could you ```rebase -I HEAD~17``` or something similar and squash the 17 commits into 1? unless there is some specific intermediate commit worth keeping.

**Review by kamilkrukowski — CHANGES_REQUESTED** [review]
1. ```generate_labels.py``` has unused parameters in the function definition. It actually uses global variables and ignores the parameters.

2. Uppercase variable names are reserved for hardcoded script-level constants. This means that args.min_occur_perc should be used instead of some uppercase version. Similar LABEL_DIR is not a hardcoded constant and violates the naming convention.

3. If we do have a command line argument or LABEL_PATH, could you keep ```labels.txt``` in the current directory by default? ie './labels.txt'

4. Could you make sure that the generate labels.txt are listed in order of how many times they occur in the training data? We'd want the most frequent label at line 1, second most at line 2, etc

**Comment by yyitingli** [comment]
> * If we do have a command line argument or LABEL_PATH, could you keep `labels.txt` in the current directory by default? ie './labels.txt'
for preprocess.py, '--labels [filename]' should it be based on current file './labels.txt 'or should it store into 'dataloader_cache/default/label.txt'?

**Comment by yyitingli** [comment]
> * `generate_labels.py` has unused parameters in the function definition. It actually uses global variables and ignores the parameters.
> * Uppercase variable names are reserved for hardcoded script-level constants. This means that args.min_occur_perc should be used instead of some uppercase version. Similar LABEL_DIR is not a hardcoded constant and violates the naming convention.
> * If we do have a command line argument or LABEL_PATH, could you keep `labels.txt` in the current directory by default? ie './labels.txt'
> * Could you make sure that the generate labels.txt are listed in order of how many times they occur in the training data? We'd want the most frequent label at line 1, second most at line 2, etc

issue fixed

**Review by kamilkrukowski — APPROVED** [review]

**Comment by kamilkrukowski** [comment]
Looks great!
